### DISCUSSION TITLE
How to group and count the elements in array simply ?

### DISCUSSION THREAD

**Discussion by BubbleGarten** [antonmedv/expr#629] [Q&A]
have :

`[1,2,2,3,3,3]`

want:
`{"1":1,"2":2,"3":"3"}`

Possible Ways: 
step1: use` groupBy()` , ` groupBy([1,2,2,3,3,3], #)` here , got:  {1:[1], 2:[2,2], 3:[3,3,3]}
step2: then `map()` ? but seems no way to built struct in expr-lang

**Comment by antonmedv** [comment]
Using groupBy can solve the problem. 

I think we need to add countBy builtin. We already have count.

**Reply by BubbleGarten** [reply]
But I don't know how to do in step2  @antonmedv , 
the second argument in ` map()` is expected to make a struct / json , but it's unsupported

trying my best:
<img width="854" alt="image" src="https://github.com/expr-lang/expr/assets/27677459/3b359507-67ca-499e-8215-fb31ab43550a">


<img width="838" alt="image" src="https://github.com/expr-lang/expr/assets/27677459/bc4958e3-77ff-461f-aab0-0befeb61508e">

<img width="952" alt="image" src="https://github.com/expr-lang/expr/assets/27677459/be40a4d8-6688-41d4-a20b-89675a85b961">
